maro/simulator/scenarios/data_center/business_engine.py

```diff
@@ -0,0 +1,253 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+from typing import List
+
+from yaml import safe_load
+
+from maro.event_buffer import AtomEvent, CascadeEvent, EventBuffer, MaroEvents
+from maro.simulator.scenarios.abs_business_engine import AbsBusinessEngine
+from maro.simulator.scenarios.helpers import DocableDict
+
+from .common import Action, DecisionEvent, Latency, VmFinishedPayload, VmRequirementPayload
+from .events import Events
+from .physical_machine import PhysicalMachine
+from .virtual_machine import VirtualMachine
+
+metrics_desc = """
+
+energy_consumption (int): Current total energy consumption.
+success_requirements (int): Accumulative successful VM requirements until now.
+failed_requirements (int): Accumulative failed VM requirements until now.
+total_latency (int): Accumulative buffer time until now.
+"""
+
+
+class DataCenterBusinessEngine(AbsBusinessEngine):
+    def __init__(
+        self, event_buffer: EventBuffer, topology: str, start_tick: int,
+        max_tick: int, snapshot_resolution: int, max_snapshots: int, additional_options: dict = {}
+    ):
+        super().__init__(
+            scenario_name="data_center", event_buffer=event_buffer, topology=topology, start_tick=start_tick,
+            max_tick=max_tick, snapshot_resolution=snapshot_resolution, max_snapshots=max_snapshots,
+            additional_options=additional_options
+        )
+
+        # Update self._config_path with current file path.
+        self.update_config_root_path(__file__)
+
+        # Env metrics.
+        self._energy_consumption: int = 0
+        self._success_requirements: int = 0
+        self._failed_requirements: int = 0
+        self._total_latency: Latency = Latency()
+
+        # Load configurations.
+        self._load_configs()
+        self._register_events()
+
+        # PM initialize.
+        self._init_machines()
+
+        # VM initialize.
+        self._vm: dict = {}
```

Comment:
How about maintaining a ``self._live_vm`` and a ``self._pending_vm_req``?

So that we can free the payload of PendingDecision Event and avoid the modification of the VM requirement outside the business engine